packages/normalizr/src/denormalize.ts

```diff
@@ -126,28 +126,33 @@ const getUnvisit = (
   ): [denormalized: any, found: boolean, deleted: boolean] {
     if (!schema) return [input, true, false];
 
-    if (!schema.denormalize || typeof schema.denormalize !== 'function') {
-      if (typeof schema === 'function') {
-        if (input instanceof schema) return [input, true, false];
-        return [new schema(input), true, false];
-      } else if (typeof schema === 'object') {
-        const method = Array.isArray(schema)
-          ? arrayDenormalize
-          : objectDenormalize;
-        return method(schema, input, unvisit);
-      }
-    }
-
     // null is considered intentional, thus always 'found' as true
     if (input === null) {
       return [input, true, false];
     }
 
+    const hasDenormalize =
+      schema.denormalize && typeof schema.denormalize === 'function';
```

Comment:
```suggestion
    const hasDenormalize = typeof schema.denormalize === 'function';
```
since if it's falsy it won't have a typeof of function